articles/multithreading-in-java/index.md

```diff
@@ -0,0 +1,168 @@
+
+1. **By extending the `Thread` class**
+
+By extending the `Thread` class, we can create an instance of a class and call the start() method:
+
+```java
+public class Main extends Thread {
+  public static void main(String[] args) {
+    Main thread = new Main();    // creates the class instance
+    thread.start();
+
+  }
+  public void run() {
+    System.out.println("This is a running thread");
+  }
+}
+```
+
+Execution of a java program starts from the `main` method. A thread is created by JVM to start the execution of the code in the `main` method and it is referred to as the main thread. The other threads are produced from the main thread and must always be the last to finish the execution.
+
+2. **By implementing the `Runnable` interface**
+
+We create instances of threads using the `new` keyword. We then specify a class that is a subclass of `Thread` .ie. `Main` as an argument to the `new` keyword. Then finally we call the `start()` method:
+
+```java
+public class Main implements Runnable {
+  public static void main(String[] args) {
+    Main obj = new Main();       // creates the class instance
+    Thread t1 = new Thread(obj); // creates the thread instance
+    t1.start();                  // starts the thread
+  }
+
+  public void run() {
+    System.out.println("This is a running thread");
+  }
+}
+```
+
+### The isAlive() method
+
+`isAlive()` method is used to check if a thread has finished executing. If the thread is still running, it returns true, otherwise false.
+
+If we use `isAlive` method before the start method, then it will print false but after the start method, it will print true.  
+Let's look at an example where we shall use the `Runnable` interface:
+
+```java
+public class Main implements Runnable {
+
+   public void run() {
+
+      Thread t = Thread.currentThread();
+      // checks if this thread is alive
+      System.out.println(t.isAlive());
+   }
+
+   public static void main(String args[]) throws Exception {
+
+      Thread t = new Thread(new Main());
+
+      // this statement will call the run() method
+      t.start();
+
+      // makes the thread to sleep for 500 milliseconds
+      t.sleep(500);
+
+      // checks if this thread is alive
+      System.out.println(t.isAlive());
+   }
+}
+```
+
+Output:
+
+```
+ true
+ false
+```
+
+### Common methods used in Thread class
+
+1. `public void run()`: Used to execute a certain task for a thread.
+2. `public void start()`: Execution of a thread starts here.
+3. `public void sleep(long milliseconds)`: Makes the executing thread sleep temporarily for a specified time.
+4. `public void join()`: Used to wait for a thread to finish its execution.
+5. `public void join(long milliseconds)`: Used to wait for a thread to finish its execution for a specified number of milliseconds.
+6. `public int getPriority()`: The priority of a thread is shown by this method.
+7. `public int setPriority(int priority)`: Used to set the thread's priority.
```

Comment:
I feel like you could have explained more on the priority in this section. If you can please add some more information on it as I believe it is important in Multithreading.